### PR TITLE
Issue 2050 gmail api extended tests

### LINKED ISSUE #2050: Extend draft test
Thinking... The above scenario looks too complex. It tests too many things that should be split. Also, the current Android realization and Espresso testing will not cover it without `dirty` code. On Android creating/updating a draft is a background task(out of the screen/Activity).

We can split the above scenario into the following
### Check the correct creating and updating
```
1. Create draft
2. Go back and create another draft
3. Check if 2 drafts are saved correctly
4. Update 1st draft. Go back to draft list screen and check if 1st draft is updated correctly and 2nd draft remains same
```

### Check the correct draft deleting
```
1. We have two drafts
2. Delete 1st draft, reload draft screen and check if 1st draft is deleted correctly. 
```

```
1. We have one drafts
2. Delete a draft and reload draft screen to check if draft folder is empty.
```

### Check the correct draft sending
```
1. Create new draft(only input subject) and go back and open that draft again
2. Update subject, insert message, recipients and click send button
3. Check if draft is sent correctly (Check if draft screen is empty after email send, go to sent folder and check if sent email is there)
```

_Originally posted by @DenBond7 in https://github.com/FlowCrypt/flowcrypt-android/pull/2046#discussion_r1016364711_

### PR DESCRIPTION
**Pull request by DenBond7**
This PR extended Gmail API draft tests

close #2050

----------------------------------

**Tests** _(delete all except exactly one)_:
- Tests added or updated

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [ ] addresses the issue it closes (if any)
- [ ] code is readable and understandable
- [ ] is accompanied with tests, or tests are not needed
- [ ] is free of vulnerabilities

### CODE REVIEW

File: FlowCrypt/src/androidTest/java/com/flowcrypt/email/ui/base/BaseDraftsGmailAPIFlowTest.kt [hunk 1]
```diff
@@ -0,0 +1,397 @@
...
+              Draft().apply {
+                id = draft.id
+                message = Message().apply {
+                  id = draft.message.id
+                }
+              }
+            }
+          }.toString()
+        )
+      }
```
Comment (ioanmo226) [inline FlowCrypt/src/androidTest/java/com/flowcrypt/email/ui/base/BaseDraftsGmailAPIFlowTest.kt:269]: I think we can combine those 2 handlers and add add condition inside combined handler?

**Review by ioanmo226 — COMMENTED** [review]
Looks great 👍 
A few small comments/questions. Please check

**Comment by DenBond7** [comment]
@ioanmo226 I've added changes. Please relook.

File: FlowCrypt/src/androidTest/java/com/flowcrypt/email/ui/DraftsGmailAPITestCorrectDeletingFlowTest.kt [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+    //swipe down to refresh
+    onView(withId(R.id.swipeRefreshLayout))
+      .perform(swipeToRefresh(swipeDown(), isDisplayingAtLeast(85)))
+    Thread.sleep(1000)
+
+    //check that there is no drafts in the local cache
+    onView(withId(R.id.recyclerViewMsgs))
+      .check(matches(withEmptyRecyclerView()))
+
+    //check that on the server side we have only one draft
```
Comment (ioanmo226) [inline FlowCrypt/src/androidTest/java/com/flowcrypt/email/ui/DraftsGmailAPITestCorrectDeletingFlowTest.kt:241]: Comment seems wrong.
`//check that on the server side we have no drafts`

**Review by ioanmo226 — APPROVED** [review]
👍 
You may update comment at next PR.

**Pull request merged**